samples/subsys/pm/power_state/src/main.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2021 SILA Embedded Solutions GmbH <office@embedded-solutions.at>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <sys/printk.h>
+#include "stm32h7xx_hal.h"
+
+void main(void)
+{
+	GPIO_InitTypeDef GPIO_InitStruct;
```

Comment:
I don't have a zephyr alternative to propose to the use of `GPIO_AF0_C1DSLEEP`.
Though, it would be better to demonstrate PM using zephyr API.
Can't `samples/boards/stm32/power_mgmt/blinky` be reused here?

Reply:
With this example I wanted to check if I am correctly switching into sleep and deep sleep, hence the specific usage of this alternate functions. Technically we could also throw the whole example away again, I basically used it only for testing. Just thought it might be useful for somebody else as well, for instance if one wants to implement other power states.
But I get the issue, it is not so nice to use HAL directly.

Reply:
Let's keep it for now if useful for testing. It's valuable for others to have something to test during review.
Point is that we'll need to rework/remove it before merge, but we're not there yet.

Reply:
Removed the example, as I do not need it anymore.